AmplifyPlugins/DataStore/AWSDataStoreCategoryPlugin/Sync/MutationSync/OutgoingMutationQueue/SyncMutationToCloudOperation.swift

```diff
@@ -67,11 +77,19 @@ class SyncMutationToCloudOperation: Operation {
                 match any known GraphQL mutation type. Ensure you only send valid mutation types:
                 \(GraphQLMutationType.allCases)
                 """
-                )
+            )
             log.error(error: dataStoreError)
+            let apiError = APIError.unknown("Invalid mutation type", "", dataStoreError)
+            finish(result: .failed(apiError))
             return
         }
 
+        if let apiRequest = createAPIRequest(mutationType: mutationType) {
+            makeAPIRequest(apiRequest)
+        }
```

Comment:
This is where the bug got introduced. Based on the comment of "bubbling up errors" i had introduced (unwittingly not noticing that we were already bubbling up the error):
```
  if let apiRequest = createAPIRequest(mutationType: mutationType) {
            makeAPIRequest(apiRequest)
+  } else {
+        finish(result: .failed(...)
+ }
```
This basically was causing us to call finish twice, which caused the unit tests to crash(because expectation's `fulfill()` calls were being called  more than expected.

So, to remedy this, i made createAPI request simpler by moving out the checking of the graphQLMutation type,  so we can easily see that the function is indeed handling the bubbling up of those errors.